File: Frontend/src/server-models/cqrs/users/requests/update-user.request.ts
import { AccountType } from 'src/server-models/enums/account-type.enum';
import { Gender } from 'src/server-models/enums/gender.enum';
import { ApplicationUser } from 'src/server-models/entities/application-user.model';

export class UpdateUserRequest {
    accountType: AccountType;
    id: string;
    username: string;
    email: string;
    firstname: string;
    lastname: string;
    gender: Gender;
    active: boolean;
}

export function GetUpdateUserRequest(user: ApplicationUser) {

    var request = new UpdateUserRequest();
    request.accountType = user.accountType;
    request.id = user.id;
    request.username = user.username;
    request.email = user.email;
    request.firstname = user.firstName;
    request.lastname = user.lastName;
    request.gender = user.gender;
    request.active = user.active;

    return request;
}

File: Backend/Core/Business/Backend.Business.Users/Users/UpdateUser/UpdateUserRequest.cs
using System;
using Backend.Domain.Entities.User;
using Backend.Domain.Enum;
using MediatR;

namespace Backend.Business.Users.Users.UpdateUser
{
    public class UpdateUserRequest : IRequest<ApplicationUser>
    {
        public Guid Id { get; set; }
        public AccountType AccountType { get; set; }

        public bool Active { get; set; }

        public string Username { get; set; }
        public string Email { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public Gender Gender { get; set; }
    }
}